optuna/visualization/_edf.py

```diff
@@ -102,30 +113,44 @@ def objective(trial, low, high):
 
     _imports.check()
 
-    if isinstance(study, Study):
-        studies = [study]
-    else:
-        studies = list(study)
+    layout = go.Layout(
+        title="Empirical Distribution Function Plot",
+        xaxis={"title": target_name},
+        yaxis={"title": "Cumulative Probability"},
+    )
 
-    _check_plot_args(studies, target, target_name)
+    info = _get_edf_info(study, target, target_name)
+    edf_lines = info.lines
 
-    return _get_edf_plot(studies, target, target_name)
+    if len(edf_lines) == 0:
+        return go.Figure(data=[], layout=layout)
+
+    traces = []
+    for study_name, y_values in edf_lines:
+        traces.append(go.Scatter(x=info.x_values, y=y_values, name=study_name, mode="lines"))
+
+    figure = go.Figure(data=traces, layout=layout)
+    figure.update_yaxes(range=[0, 1])
+
+    return figure
 
 
-def _get_edf_plot(
-    studies: List[Study],
+def _get_edf_info(
+    study: Union[Study, Sequence[Study]],
     target: Optional[Callable[[FrozenTrial], float]] = None,
     target_name: str = "Objective Value",
-) -> "go.Figure":
-    layout = go.Layout(
-        title="Empirical Distribution Function Plot",
-        xaxis={"title": target_name},
-        yaxis={"title": "Cumulative Probability"},
-    )
+) -> _EDFInfo:
+
+    if isinstance(study, Study):
+        studies = [study]
+    else:
+        studies = list(study)
+
+    _check_plot_args(studies, target, target_name)
 
     if len(studies) == 0:
         _logger.warning("There are no studies.")
-        return go.Figure(data=[], layout=layout)
+        return _EDFInfo(lines=[], x_values=np.array([]))
```

Comment:
This is a small nit, but maybe `lines` and `x_values` should be `Optional`? This would simplify checking for empty plot in concrete implementations and i guess reduce allocation a bit. What do you think @nzw0301?

Reply:
Thank you for your comment and concrete suggestions! I think the other visualisation parts (ex. #3682) might have a similar case; @HideakiImamura do you have a preference? 

Explicitly returning `lines=None, x_values=None` sounds easy to test the logic when plot data is invalid. One drawback might be to add a few lines to avoid mypy error.

Reply:
Thank you for the opinion, let's keep it as it is then.

Reply:
In fact, I realised that we might be able to simplify the initialisation by using the default values as in https://github.com/optuna/optuna/pull/3781.